src/objects/Control.ts

```diff
@@ -0,0 +1,66 @@
+// keyboard, mouse and touch controller!
+
+class Control {
+  private scene: Phaser.Scene;
+
+  private zones: Record<string, number>;
+
+  private key: Record<string, Phaser.Input.Keyboard.Key> | undefined;
+
+  public balance = 0;
+
+  public jump = false;
+
+  constructor(scene: Phaser.Scene) {
+    this.scene = scene;
+
+    const { width } = scene.sys.game.canvas;
+    this.zones = {
+      halfWidth: width / 2, // half screen.
+      quarterWidth: width / 4, // quarter screen.
+      deadZone: width - 100,
+    };
+
+    // all devices have pointer (mouse or touchscreen)
+    // some devices have keyboards
+    // we register keys on mount
+    const { keyboard } = scene.input;
+    if (keyboard) {
+      this.key = {
+        left: keyboard.addKey('A'),
+        right: keyboard.addKey('D'),
+        jump: keyboard.addKey('SPACE'),
+      };
+    }
+  }
+
+  // this needs to be called in the scene update (every frame)
+  update() {
+    const { isDown, x } = this.scene.input.activePointer;
+    const { quarterWidth, halfWidth, deadZone } = this.zones;
+
+    // touch booleans
+    const isPointerLeft = x < quarterWidth && isDown;
+    const isPointerRight = x >= quarterWidth && x < halfWidth && isDown;
+    const isPointerJump = x >= halfWidth && x < deadZone && isDown;
+
+    // keyboard booleans
+    const isKeyLeft = this.key?.left.isDown || false;
```

Comment:
hmmm so we're not actually listening to events for keys with this? guess this would solve the laggy issue where we need to remove the listeners on skepta shutdown